wallet-new/test/unit/UTxO/PreChain.hs

```diff
@@ -64,29 +73,82 @@ preChain = fmap (, ()) . DepIndep
 -- See 'fromPreChain'
 data FromPreChain h a = FromPreChain {
       -- | The boot transaction that was used in the translation
-      fpcBoot :: Transaction h Addr
+      fpcBoot   :: Transaction h Addr
 
       -- | The resulting chain (i.e., list of list of transactions)
       -- This does /not/ include the bootstrap transaction.
-    , fpcChain :: Chain h Addr
+    , fpcChain  :: Chain h Addr
 
       -- | The resulting ledger (i.e., flat list of transactions)
       -- This /does/ include the bootstrap transaction .
     , fpcLedger :: Ledger h Addr
 
       -- | Any additional information that was included in the 'PreChain'.
-    , fpcExtra :: a
+    , fpcExtra  :: a
     }
 
 fromPreChain :: (Hash h Addr, Monad m)
              => PreChain h m a -> TranslateT IntException m (FromPreChain h a)
 fromPreChain pc = do
     fpcBoot <- asks bootstrapTransaction
-    (txs, fpcExtra) <- calcFees fpcBoot =<< lift (runDepIndep pc fpcBoot)
+    (txs, fpcExtra) <- calcChainFees fpcBoot =<< lift (runDepIndep pc fpcBoot)
     let fpcChain  = Chain txs -- doesn't include the boot transactions
         fpcLedger = chainToLedger fpcBoot fpcChain
     return FromPreChain{..}
 
+{-------------------------------------------------------------------------------
+  Tree with some information still missing.
+-------------------------------------------------------------------------------}
+
+-- | Blocks organised in a tree structure.
+type BlockTree h a = OldestFirst Tree (UTxO.DSL.Block h a)
+
+-- | Generalisation of a 'PreChain'.
+--   A 'PreChain' is the flattening of a 'PreTree' with a single branch.
+type PreTree h m a = DepIndep (Transaction h Addr) (Tree [Fee]) m (BlockTree h Addr, a)
```

Comment:
Are there some guidelines about type synonyms like this? My main concern is that many times the rather cryptic expanded type synonym ends up showing up in error messages from the compiler, and that makes maintaining and refactoring the code much harder.